scripts/run_mesh_experiments.py

```diff
@@ -0,0 +1,293 @@
+"""
+Experiments on disease mesh tags to understand role of params
+"""
+from argparse import ArgumentParser
+from datetime import datetime
+from pathlib import Path
+import json
+import math
+
+from wellcomeml.ml import KerasVectorizer, CNNClassifier, BiLSTMClassifier
+from tensorboard.plugins.hparams import api as hp
+from sklearn.preprocessing import MultiLabelBinarizer
+from sklearn.metrics import f1_score
+from scipy.sparse import vstack, csr_matrix
+from tqdm import tqdm
+import tensorflow as tf
+import numpy as np
+
+
+def create_dataset(data_path, nb_tags, nb_examples_per_tag):
+    texts = []
+    tags = []
+
+    tags_count = {}
+    with open(data_path) as f:
+        for line in f:
+            item = json.loads(line)
+            item_tags = []
+            for tag in item["tags"]:
+                if tag not in tags_count:
+                    if len(tags_count) < nb_tags:
+                        tags_count[tag] = 1
+                        item_tags.append(tag)
+                    else:
+                        # tags full
+                        pass
+                else: # tag in tags count
+                    tags_count[tag] += 1
+                    item_tags.append(tag)
+            if item_tags:
+                texts.append(item["text"])
+                tags.append(item_tags)
+            if all([c > nb_examples_per_tag for t, c in tags_count.items()]):
+                break
+        return texts, tags
+
+def vectorize_data(train_texts, train_tags, test_texts, test_tags,
+                   vocabulary_size, sequence_length):
+    # fit vectorizer and transform
+    vectorizer = KerasVectorizer(vocabulary_size, sequence_length)
+    X_train = vectorizer.fit_transform(train_texts)
+    X_test = vectorizer.transform(test_texts)
+
+    # vectorize tags
+    label_binarizer = MultiLabelBinarizer(sparse_output=True)
+    label_binarizer.fit(train_tags)
+    Y_train = label_binarizer.transform(train_tags)
+    Y_test = label_binarizer.transform(test_tags)
+    return X_train, X_test, Y_train, Y_test
+
+def build_cnn_model(learning_rate=0.01, batch_size=256, attention=True,
+                    vocabulary_size=400_000, sequence_length=400, nb_tags=512,
+                    l2=1e-6, dropout=0.1, hidden_size=100, dense_size=100_000):
```

Comment:
dense_size default should be 10k :p

Reply:
you mean the 100_000. it is python notation to make it more readable. https://www.python.org/dev/peps/pep-0515/